### PR TITLE
feat(date-picker): add Jalali (Persian) calendar support

### PR DESCRIPTION
**Pull request by mehrabix**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our
      guidelines: https://github.com/spartan-ng/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### Primitives

- [ ] accordion
- [ ] alert
- [ ] alert-dialog
- [ ] aspect-ratio
- [ ] autocomplete
- [ ] avatar
- [ ] badge
- [ ] breadcrumb
- [ ] button
- [ ] button-group
- [x] calendar
- [ ] card
- [ ] carousel
- [ ] checkbox
- [ ] collapsible
- [ ] combobox
- [ ] command
- [ ] context-menu
- [ ] data-table
- [x] date-picker
- [ ] dialog
- [ ] empty
- [ ] dropdown-menu
- [ ] field
- [ ] hover-card
- [ ] icon
- [ ] input
- [ ] input-group
- [ ] input-otp
- [ ] item
- [ ] kbd
- [ ] label
- [ ] menubar
- [ ] native-select
- [ ] navigation-menu
- [ ] pagination
- [ ] popover
- [ ] progress
- [ ] radio-group
- [ ] resizable
- [ ] scroll-area
- [ ] select
- [ ] separator
- [ ] sheet
- [ ] sidebar
- [ ] skeleton
- [ ] slider
- [ ] sonner
- [ ] spinner
- [ ] switch
- [ ] table
- [ ] tabs
- [ ] textarea
- [ ] toggle
- [ ] toggle-group
- [ ] tooltip
- [ ] typography

### Others

- [ ] trpc
- [ ] nx
- [ ] repo
- [ ] cli

## What is the current behavior?

The date-picker and calendar components only support the Gregorian calendar. There is no built-in support for Jalali (Persian/Solar Hijri) dates, which is the official calendar of Iran and Afghanistan.

## What is the new behavior?

Adds a complete Jalali (Persian) calendar date picker using pure math algorithms — no external npm packages required. Includes:

- **`BrnJalaliDateAdapter`** — implements `BrnDateAdapter<JalaliDate>` with proper Jalali month arithmetic (fixed: month wrap formula now uses Euclidean modulo to avoid invalid month 13 when subtracting exact multiples of 12), day-by-day iteration via Gregorian round-trip, and 0-indexed Jalali month access
- **`JalaliDate`** — date class with `valueOf()` returning Gregorian timestamp for reliable `<=` comparison used by the calendar
- **Persian calendar math** — pure Borkowski algorithms for `jalaliToGregorian`, `gregorianToJalali`, `getDaysInMonth`, `isLeapJalaliYear`
- **Persian i18n** — Farsi weekday/month names, RTL layout, Saturday as first day of week
- **Docs pages** — live "Jalali (Persian) Calendar" / "Jalali (Persian) Date Picker" examples with inline code on both calendar and date-picker docs pages
- **Storybook story** — Persian locale date picker with dropdown caption layout
- **Language switcher** — "Persian (فارسی)" added to the RTL preview language toggle, with `'fa'` support in `TranslateService`
- **51 unit tests** — covering every adapter method with known Jalali↔Gregorian conversions, including edge cases for month wrap (subtract/add multiples of 12 months)
- **Translation fallback** — all 29 RTL preview components use `?? this._translations['en']!` fallback to prevent runtime crash when `'fa'` is selected
- **Interface compliance** — `getHours`/`getMinutes`/`getSeconds`/`getMilliseconds` now include the required `date: JalaliDate` parameter

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

The implementation uses the well-known algorithm by Kazimierz M. Borkowski for Jalali↔Gregorian conversion, which is already battle-tested in the user's standalone `persian-datepicker-element` project. No external dependencies are introduced.

## Summary by CodeRabbit

* **New Features**
  * Persian (Jalali) calendar support: new Jalali date type, adapter, calendar and date-picker examples, and Storybook story; UI shows Jalali formatting.
  * Persian (fa) added to language selector and UI previews.

* **Bug Fixes**
  * RTL preview components now reliably fall back to English translations to avoid missing text/direction.

* **Tests**
  * Unit tests added for the Jalali date adapter and utilities.

### CODE REVIEW

**Comment by mehrabix** [comment]
@marcjulian @MerlinMoos @ashley-hunter @goetzrobin would you please review this?

**Comment by MerlinMoos** [comment]
hey @mehrabix LGTM thanks for the addition, can we revert all the language support on the components, and add this example to our blocks page? You can add a new datepicker block page or just add it to the https://spartan.ng/blocks/calendar

File: apps/app/src/app/pages/(components)/components/(accordion)/accordion--rtl.example.ts [hunk 1]
```diff
@@ -88,7 +88,7 @@ export class AccordionRtl {
 		},
 	];
 
-	private readonly _translation = computed(() => this._translations[this._language()]);
+	private readonly _translation = computed(() => this._translations[this._language()] ?? this._translations['en']!);
```
Comment (MerlinMoos) [inline apps/app/src/app/pages/(components)/components/(accordion)/accordion--rtl.example.ts]: after adding the example to the blocks please revert all of this lines of codes in the rtl examples

Comment (mehrabix) [reply apps/app/src/app/pages/(components)/components/(accordion)/accordion--rtl.example.ts]: Hi, Done! Reverted all the ?? this._translations['en']! lines across the RTL examples and cleaned up the translate service / language picker. Removed the Jalali examples from the calendar & date-picker docs pages and added it as a block under /blocks/calendar instead.

File: libs/brain/date-time/src/lib/jalali-date-adapter.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	}
+
+	set(date: JalaliDate, values: BrnDateUnits): JalaliDate {
+		const y = values.year ?? date.year;
+		const m = values.month !== undefined ? values.month + 1 : date.month;
+		const d = values.day ?? date.day;
+		return this._constrain(y, m, d);
+	}
+
+	add(date: JalaliDate, duration: BrnDuration): JalaliDate {
```
Comment (goetzrobin) [inline libs/brain/date-time/src/lib/jalali-date-adapter.ts:29]: When both months and days are passed in one call, the day isn't clamped between the two steps, so it can drift by a day. E.g. add(1400/6/31, { months: 1, days: 5 }): after the month step we have an intermediate 1400/7/31, but Mehr only has 30 days - so jalaliToGregorian(1400, 7, 31) rolls into Aban 1 and the +5 is measured from there, giving 1400/8/6 instead of the clamp-then-add result 1400/8/5. The trailing _constrain can't catch it because the overflow already happened inside the round-trip.

File: libs/brain/date-time/src/lib/persian-date.ts
Comment (goetzrobin) [inline libs/brain/date-time/src/lib/persian-date.ts:1]: PersianDate is currently only exercised indirectly through the adapter. Since it's exported as a standalone utility, it deserves its own direct tests — jalaliToGregorian/gregorianToJalali round-trips against known dates, getDaysInMonth for the 31/30/29/30-day cases, and isLeapJalaliYear across known leap/non-leap years. Note the leap rule is the 33-year arithmetic cycle, which matches ICU/CLDR but can differ from the astronomical calendar in disputed years like 1403 - worth a one-line comment.

File: libs/brain/date-time/src/lib/jalali-date.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { PersianDate } from './persian-date';
+
+export class JalaliDate {
+	constructor(
+		public readonly year: number,
+		public readonly month: number,
+		public readonly day: number,
+	) {}
+
+	valueOf(): number {
```
Comment (goetzrobin) [inline libs/brain/date-time/src/lib/jalali-date.ts:10]: The same terse locals as in PersianDate. (valueOf() does const g = …; new Date(g[0], g[1] - 1,
  g[2])) and the adapter (getDay, add, now use g/y/m). Expand those to descriptive names, or at least destructure const
  [gregorianYear, gregorianMonth, gregorianDay] = … so the tuple indexing reads clearly.

File: libs/brain/date-time/src/lib/persian-date.ts
Comment (goetzrobin) [reply libs/brain/date-time/src/lib/persian-date.ts:1]: snake_case → camelCase, and expand the cryptic j/g prefixes: j_days_in_month → jalaliDaysInMonth, g_days_in_month → gregorianDaysInMonth, params j_y/j_m/j_d → jalaliYear/jalaliMonth/jalaliDay, g_y/g_m/g_d → gregorianYear/gregorianMonth/gregorianDay.

File: libs/brain/date-time/src/lib/persian-date.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
+export const PersianDate = {
+	g_days_in_month: [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31],
+	j_days_in_month: [31, 31, 31, 31, 31, 31, 30, 30, 30, 30, 30, 29],
+
+	jalaliToGregorian: function (j_y: number, j_m: number, j_d: number): [number, number, number] {
+		j_y = parseInt(j_y.toString());
```
Comment (goetzrobin) [inline libs/brain/date-time/src/lib/persian-date.ts]: parseInt(j_y.toString()) is a no-op on already-typed number params and can be dropped.

File: libs/brain/date-time/src/lib/persian-date.ts [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+		}
+
+		const gm = i + 1;
+		const gd = gDayNo + 1;
+
+		return [gy, gm, gd];
+	},
+
+	gregorianToJalali: function (g_y: number, g_m: number, g_d: number): [number, number, number] {
+		g_y = parseInt(g_y.toString());
```
Comment (goetzrobin) [inline libs/brain/date-time/src/lib/persian-date.ts]: parseInt(j_y.toString()) is a no-op on already-typed number params and can be dropped.

File: libs/brain/date-time/src/lib/persian-date.ts [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+	},
+
+	getDaysInMonth: function (year: number, month: number): number {
+		if (month < 1 || month > 12) return 0;
+		if (month <= 6) return 31;
+		if (month <= 11) return 30;
+		return this.isLeapJalaliYear(year) ? 30 : 29;
+	},
+
+	getMonthName: function (month: number): string {
```
Comment (goetzrobin) [inline libs/brain/date-time/src/lib/persian-date.ts]: I think it is fine to keep the numeric math helpers (getDayOfWeek, getDaysInYear, isValidDate) as a convenience surface, but getMonthName returns a localized Persian string — that's an i18n/presentation concern that already lives in the calendar i18n (component + story). Suggest dropping it from the lib to keep PersianDate purely calendrical.

File: apps/app/src/app/pages/(blocks)/blocks/calendar/calendar-jalali/calendar-jalali.component.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+			'مهر',
+			'آبان',
+			'آذر',
+			'دی',
+			'بهمن',
+			'اسفند',
+		] as MonthLabels,
+	years: (startYear = 1300, endYear = 1420) => Array.from({ length: endYear - startYear + 1 }, (_, i) => startYear + i),
+	formatHeader: (month, year) =>
+		['فروردین', 'اردیبهشت', 'خرداد', 'تیر', 'مرداد', 'شهریور', 'مهر', 'آبان', 'آذر', 'دی', 'بهمن', 'اسفند'][month] +
```
Comment (goetzrobin) [inline apps/app/src/app/pages/(blocks)/blocks/calendar/calendar-jalali/calendar-jalali.component.ts]: months arrays are repeated 3 times. can we turn into one shared const?

File: apps/ui-storybook/stories/date-picker-jalali.stories.ts [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+			'شهریور',
+			'مهر',
+			'آبان',
+			'آذر',
+			'دی',
+			'بهمن',
+			'اسفند',
+		] as MonthLabels,
+	years: (startYear = 1300, endYear = 1420) => Array.from({ length: endYear - startYear + 1 }, (_, i) => startYear + i),
+	formatHeader: (month, year) =>
```
Comment (goetzrobin) [inline apps/ui-storybook/stories/date-picker-jalali.stories.ts]: months arrays are repeated 3 times. can we turn into one shared const?

**Review by goetzrobin — CHANGES_REQUESTED** [review]
Looks great! Some minor improvements that get this over the finish line!

File: libs/brain/date-time/src/lib/jalali-date-adapter.ts [hunk 9]
```diff
@@ -0,0 +1,156 @@
+import type { BrnDateAdapter, BrnDateUnits, BrnDuration } from './date-adapter';
+import { JalaliDate } from './jalali-date';
+import { PersianDate } from './persian-date';
+
+export class BrnJalaliDateAdapter implements BrnDateAdapter<JalaliDate> {
+	create({ year, month, day }: BrnDateUnits): JalaliDate {
+		const now = this.now();
+		const m = month !== undefined ? month + 1 : now.month;
+		return new JalaliDate(year ?? now.year, m, day ?? now.day);
```
Comment (goetzrobin) [inline libs/brain/date-time/src/lib/jalali-date-adapter.ts]: should  `_constrain` be used here?

File: libs/brain/date-time/src/lib/jalali-date-adapter.ts [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+	}
+
+	now(): JalaliDate {
+		const d = new Date();
+		const [jalaliYear, jalaliMonth, jalaliDay] = PersianDate.gregorianToJalali(
+			d.getFullYear(),
+			d.getMonth() + 1,
+			d.getDate(),
+		);
+		return new JalaliDate(jalaliYear, jalaliMonth, jalaliDay);
```
Comment (goetzrobin) [inline libs/brain/date-time/src/lib/jalali-date-adapter.ts]: should  `_constrain` be used here?

File: libs/brain/date-time/src/lib/persian-date.ts [hunk 11]
```diff
@@ -0,0 +1,118 @@
+export const PersianDate = {
+	gregorianDaysInMonth: [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31],
+	jalaliDaysInMonth: [31, 31, 31, 31, 31, 31, 30, 30, 30, 30, 30, 29],
+
+	jalaliToGregorian: function (jalaliYear: number, jalaliMonth: number, jalaliDay: number): [number, number, number] {
+		const jy = jalaliYear - 979;
```
Comment (goetzrobin) [inline libs/brain/date-time/src/lib/persian-date.ts]: jy is confusing because one would assume it stands for jalaliYear but we subrtract 979 from it, similar for jm and jd

File: libs/brain/date-time/src/lib/persian-date.ts [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+
+		return [gy, gm, gd];
+	},
+
+	gregorianToJalali: function (
+		gregorianYear: number,
+		gregorianMonth: number,
+		gregorianDay: number,
+	): [number, number, number] {
+		const gy = gregorianYear - 1600;
```
Comment (goetzrobin) [inline libs/brain/date-time/src/lib/persian-date.ts]: same here

**Review by goetzrobin — CHANGES_REQUESTED** [review]
couple more naming issues and questions about `_constrain`

File: libs/brain/date-time/src/lib/persian-date.ts (hunk 12 above)
Comment (mehrabix) [reply libs/brain/date-time/src/lib/persian-date.ts]: Already renamed in the latest commit — gy → gregorianYearDelta, along with gm → gregorianMonthIdx and gd → gregorianDayIdx throughout persian-date.ts.

File: libs/brain/date-time/src/lib/jalali-date-adapter.ts (hunk 9 above)
Comment (mehrabix) [reply libs/brain/date-time/src/lib/jalali-date-adapter.ts]: I think you're right. now(): Same issue. gregorianToJalali returns mathematically correct values, but in edge cases (e.g., near midnight with timezone drift), the conversion could theoretically produce a day that exceeds the month's max. _constrain is a safe guard. The set() and add() methods already use _constrain — create() and now() were the only two paths that skipped it.

File: libs/brain/date-time/src/lib/jalali-date-adapter.ts (hunk 10 above)
Comment (mehrabix) [reply libs/brain/date-time/src/lib/jalali-date-adapter.ts]: I think you're right. now(): Same issue. gregorianToJalali returns mathematically correct values, but in edge cases (e.g., near midnight with timezone drift), the conversion could theoretically produce a day that exceeds the month's max. _constrain is a safe guard. The set() and add() methods already use _constrain — create() and now() were the only two paths that skipped it.

File: libs/brain/date-time/src/lib/persian-date.ts (hunk 11 above)
Comment (mehrabix) [reply libs/brain/date-time/src/lib/persian-date.ts]: Already renamed in the latest commit — gy → gregorianYearDelta, along with gm → gregorianMonthIdx and gd → gregorianDayIdx throughout persian-date.ts.

**Review by MerlinMoos — APPROVED** [review]

File: libs/brain/date-time/src/lib/persian-date.ts [hunk 13]
```diff
@@ -0,0 +1,153 @@
+export const PersianDate = {
+	gregorianDaysInMonth: [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31],
+	jalaliDaysInMonth: [31, 31, 31, 31, 31, 31, 30, 30, 30, 30, 30, 29],
+
+	jalaliToGregorian: function (jalaliYear: number, jalaliMonth: number, jalaliDay: number): [number, number, number] {
+		jalaliYear = parseInt(jalaliYear.toString());
```
Comment (goetzrobin) [inline libs/brain/date-time/src/lib/persian-date.ts]: the parseInt(.toString())'s should be removed since the function inputs are typed as numbers

File: libs/brain/date-time/src/lib/persian-date.ts
Comment (goetzrobin) [inline libs/brain/date-time/src/lib/persian-date.ts:1]: can we add unit tests for this since its exported as a standalone utility?

File: libs/brain/date-time/src/lib/persian-date.ts (hunk 6 above)
Comment (goetzrobin) [reply libs/brain/date-time/src/lib/persian-date.ts]: we should drop this function, because it is an i18n concern, not logic

**Review by goetzrobin — CHANGES_REQUESTED** [review]
Couple more things. Please add tests for PersianDate, remove getMonthName, and get rid of the parseInt(xxx.toString()) when we already require numbers to be passed in as parameters in the function signature
